packages/backend/src/matter/behaviors/color-control-server.ts

```diff
@@ -174,6 +142,28 @@ export class ColorControlServerBase extends FeaturedBase {
     }
     return undefined;
   }
+
+  private getMatterColorMode(
+    mode: LightDeviceColorMode | undefined,
+  ): ColorControl.ColorMode {
+    // This cluster is only used with HueSaturation, ColorTemperature or Both.
+    // It is never used without any of them.
+    if (this.features.colorTemperature && this.features.hueSaturation) {
+      if (mode === LightDeviceColorMode.COLOR_TEMP) {
+        return ColorControl.ColorMode.ColorTemperatureMireds;
+      } else {
+        return ColorControl.ColorMode.CurrentHueAndCurrentSaturation;
+      }
+    } else if (this.features.colorTemperature) {
+      return ColorControl.ColorMode.ColorTemperatureMireds;
```

Comment:
The idea of the previous setup was to let matter.js state be (which will internally transition around on varios commansd) if the light entity wasn't in a supported color mode, but I suppose in retrospect that it doesn't make much sense as we'd still be messing with the `current*` state anyway...